docs/metrics/customizing-the-sdk/Program.cs

```diff
@@ -0,0 +1,99 @@
+// <copyright file="Program.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics.Metrics;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+
+public class Program
+{
+    private static readonly Meter Meter1 = new Meter("CompanyA.ProductA.Library1", "1.0");
+    private static readonly Meter Meter2 = new Meter("CompanyA.ProductB.Library2", "1.0");
+
+    public static void Main(string[] args)
+    {
+        using var meterProvider = Sdk.CreateMeterProviderBuilder()
+            .AddSource(Meter1.Name)
+            .AddSource(Meter2.Name)
+
+            // Rename an instrument to new name.
+            .AddView(instrumentName: "MyCounter", name: "MyCounterRenamed")
+
+            // Change Histogram bounds
+            .AddView(instrumentName: "MyHistogram", new HistogramConfig() { HistogramBounds = new double[] { 10, 20 } })
+
+            // For the instrument "MyCounterCustomTags", aggregate with only the keys "tag1", "tag2".
+            .AddView(instrumentName: "MyCounterCustomTags", new AggregationConfig() { TagKeys = new string[] { "tag1", "tag2" } })
+
+            // Drop the instrument "MyCounterDrop".
+            .AddView(instrumentName: "MyCounterDrop", new DropAggregationConfig())
+
+            // Advanced selection criteria and config via Func<Instrument, AggregationConfig>
+            .AddView((instrument) =>
+             {
+                 if (instrument.Meter.Name.Equals("CompanyA.ProductB.Library2") &&
+                     instrument.GetType().Name.StartsWith("Histogram"))
```

Comment:
might benefit from some extension method like instrument.IsHistogram?

Reply:
I like this idea - fine as a follow up probably.

Maybe something like `instrument.InstrumentType` that returns an enum'ed value.